Documentation/Architecture/InputSystem.md

```diff
@@ -0,0 +1,66 @@
+# Input System
+
+The input system and is the largest and most interesting system out of all the features offered by the MRTK.
+So many things within the toolkit build on top of it (pointers, focus, prefabs). The code within the input
+system is what allows for the natural interactions like grab and rotate.
+
+The input system has some of its own terminology that are worth defining:
+
+- **Data Providers**
+
+    The input settings in the input profile have references to entities known as "data providers" - another word
+    that describes these are “device managers”. These are components whose job is to extend the MRTK input system
+    by interfacing with a specific underlying system. An example of a provider is the Windows Mixed Reality provider,
+    whose job it is to talk with the underlying Windows Mixed Reality APIs, and then translate the data from
+    those APIs into MRTK-specific input concepts below. Another example would be the OpenVR provider (who job it
+    is to talk to Unity-abstracted version of OpenVR APIs and then translate that data into MRTK input concepts).
+
+- **Controller**
+
+    A representation of a physical controller (whether it’s a 6-degree-of-freedom controller, a HoloLens 1-style
+    hand with gesture support, a fully articulated hand, a leap motion controller, etc.). Controllers are spawned
+    by device managers (i.e. the WMR device manager will spawn a controller and manage its lifetime when it sees an
+    articulated hand coming into existence, for example).
+
+- **Pointer**
+
+    In order to interact with the rest of the world, a controller is associated with pointers, which can interact with
+    other game objects in specific ways. For example, the near interaction pointer is responsible to detecting when
+    your hand (which is a controller) is close to objects that advertise themselves as supporting ‘near interaction’.
+    There are pointers that handle teleportation. There are far pointers (i.e. the shell hand ray pointer) that uses
+    far raycasts to engage with content that is longer than arms length from the user.
+
+    Note that a controller can be associated with many different pointers at the same time – in order to ensure that
+    this doesn’t devolve into chaos, there is also a pointer mediator which controls which pointers are allowed to be
+    active (for example, this will disable far interaction pointers when near interaction is detected).
+
+    Pointers can also belong to different categories like near interaction (useful for dealing with objects that are
```

Comment:
i think we already had the definition before where we're saying an arm length away from the user is far interaction - i would remove this sentence?